pkg/mutation/path/parser/node.go

```diff
@@ -112,3 +138,23 @@ func (l List) Value() (string, bool) {
 	}
 	return *l.KeyValue, true
 }
+
+func (l List) String() string {
+	key := quote(l.KeyField)
+	if l.Glob {
+		return fmt.Sprintf("[%s: *]", key)
+	}
+	if l.KeyValue != nil {
+		value := quote(*l.KeyValue)
+		return fmt.Sprintf("[%s: %s]", key, value)
+	}
+	// Represents an improperly specified List node.
+	return fmt.Sprintf("[%q: ]", key)
+}
+
+// quote adds double quotes around the passed string.
+func quote(s string) string {
+	// Using fmt.Sprintf with %q converts whitespace to escape sequences, and we
+	// don't want that.
+	return `"` + s + `"`
```

Comment:
Do we need to ensure that all `"` strings in s are escaped?

Reply:
Sounds good. Also added test.

Reply:
This only applies to identifiers which include escaped quotes.